include/toolchain/gcc.h

```diff
@@ -94,6 +94,11 @@ do {                                                                    \
 #define __weak __attribute__((__weak__))
 #define __unused __attribute__((__unused__))
 
+/* Be *very* careful with this, you cannot filter out with -wno-deprecated,
+ * which has implications for -Werror
+ */
+#define __DEPRECATED_MACRO _Pragma("GCC warning \"Macro is deprecated\"")
```

Comment:
What you're proposing here isn't possible with the C preprocessor, you can't implement ZEPHYR_VERSION_CHECK(maj, min) in this way.
Best I can do is a #if ... #else on the SW version at the __stack definition I am trying to deprecate.